### ISSUE TITLE
Version 1.8.1 breaks i Meteor.js when using nvd3:nvd3 package.

### ISSUE THREAD

**Issue by JesperWe** [novus/nvd3#1185]
I did a "meteor update" and got v1.8.1 of the nvd3:nvd3 package.
My application broke, and it turns out the global nv variable is === undefined.
I reverted to 1.7.1 and everything works again.

**Comment by liquidpele** [comment]
I'm not familiar with meteor . ..   Can you look at it and send a pull request to fix?   Latest code is in master branch

**Comment by JesperWe** [comment]
So who creates the nvd3 package (https://atmospherejs.com/nvd3/nvd3) ??
It has no maintainer visible except "nvd3", which makes it look like an official package from the nvd3 project....

**Comment by mvanio** [comment]
With meteor 1.1.0.3 and nvd3 v 1.7.0/1.7.1/1.8.1 I'm seeing this issue also.

    Exception from Tracker afterFlush function:
    debug.js:41 TypeError: Cannot read property 'addGraph' of undefined
    at Template.theChart.rendered (chart_data.js?1eb885b491def976b3d4d08c8d073ec3d2bba36e:79)
        at template.js:116
        at Function.Template._withTemplateInstanceFunc (template.js:437)
        at fireCallbacks (template.js:112)
        at null.<anonymous> (template.js:205)
        at view.js:104
        at Object.Blaze._withCurrentView (view.js:523)
        at view.js:103
        at Object.Tracker._runFlush (tracker.js:468)
        at onGlobalMessage (setimmediate.js:102)

**Comment by gallen** [comment]
Same issue here. Is there anyone checking this?

**Comment by liquidpele** [comment]
I don't know anything about meteor...

**Comment by gallen** [comment]
It seems caused by 'var' when define nv.
  // set up main nv object 
  var nv = {};               

Removing 'var' in front of nv works. Meteor packaging system wraps the entire file to an anonymous function, and 'nv' was not in global anymore if 'var' is there.

**Comment by liquidpele** [comment]
Hmm, previously it was:

    var nv = window.nv || {};
    window.nv = nv;

However, we undid that because it fails if loaded in shadow dom since there isn't a window yet in that case... guess we'll need to find a way to make it work in all cases...   perhaps this?

    var nv = {};
    if (window) {
        nv = window.nv || {};
        window.nv = nv;
    }
    
Seems like it was changed in this commit:  https://github.com/novus/nvd3/commit/52846bacfa0960fb19b2a78ae72fc09973e9e920#diff-7eb52b366866677666470e019283c8ea

@jjzazuet can you comment on why you made that change?

**Comment by mayvn10** [comment]
I'm running Meteor 1.1.0.3.

Just tried downgrading to nvd3:nvd3 1.7.1 - that didn't work.

Then I tried declaring ```var nv = {}``` in both the downgraded version and the updated version - that didn't work.

Any updates here? Is anyone in contact with the package container?

**Comment by timhoeppner** [comment]
Downgrading works for me, what does "meteor list" spit out for your project?

**Comment by mayvn10** [comment]
This is what I get:

```
Meteor 1.1.0.3
```

```
accounts-base                 1.2.0  A user account system
accounts-password             1.1.1  Password support for accounts
arkham:comments-ui            0.2.10  Simple templates for disqus-like commen...
chrismbeckett:toastr          2.1.2_1  Gnome / Growl type non-blocking notifi...
d3js:d3                       3.5.5  D3 (official): A JavaScript visualizatio...
email                         1.0.6  Send email messages
gliese:pretty-email           0.1.2  Send pretty emails (fork:hashbang & coff...
http                          1.1.0  Make HTTP calls to remote servers
iron:location                 1.0.9  Reactive urls that work in IE8/9 and mod...
iron:router                   1.0.9  Routing specifically designed for Meteor
kevohagan:sweetalert          0.5.0* a beautiful replacement for javascript's...
manuelschoebel:ms-seo         0.4.1  Easily config SEO for your routes
matteodem:easy-search         1.6.3  Easy-to-use search with Blaze Components...
meteor-platform               1.2.2  Include a standard set of Meteor package...
meteorhacks:aggregate         1.3.0  Proper MongoDB aggregations support for ...
momentjs:moment               2.10.6  Moment.js (official): parse, validate, ...
mongo-livedata                1.0.8  Moved to the 'mongo' package
msavin:mongol                 1.1.5  The insanely handy development package f...
nvd3:nvd3                     1.7.1* Nvd3.org charts.
percolate:momentum            0.7.2  Reactive animations
reactive-var                  1.0.5  Reactive variable
reywood:publish-composite     1.4.2  Publish a set of related documents from ...
service-configuration         1.0.4  Manage the configuration for third-party...
```

**Comment by timhoeppner** [comment]
Here's mine, same meteor version:

```
d3js:d3          3.5.5  D3 (official): A JavaScript visualization library for...
iron:router      1.0.9  Routing specifically designed for Meteor
meteor           1.1.6  Core Meteor environment
meteor-platform  1.2.2  Include a standard set of Meteor packages in your app
nvd3:nvd3        1.7.1* Nvd3.org charts.
twbs:bootstrap   3.3.5  The most popular front-end framework for developing r...
```

**Comment by mayvn10** [comment]
Question - what's the difference between Meteor 1.1.0.3 and your meteor 1.1.6? Maybe that's what's missing from mine and why it's not working?

**Comment by timhoeppner** [comment]
meteor --version gives 1.1.0.3

I might have added the meteor package at some point when I was trying to get this to work. I'll try removing it and see if it still works

**Comment by mayvn10** [comment]
Ok maybe it'll help if I show you the code that I have in the rendered block of Meteor that's causing the addGraph error:

```
nv.addGraph(function() {
	chart = nv.models.lineChart()
		.margin({ left: 25 })
		.useInteractiveGuideline(true)
		.transition()
		.duration(350)
		.showLegend(true)
		.showYAxis(true)
		.showXAxis(true)
		.options({
			transitionDuration: 300,
			useInteractiveGuideline: true
		});

	chart.xAxis
		.axisLabel('Date')
		.tickFormat(d3.format('d'));

	chart.yAxis
		.axisLabel('Tracks')
		.tickFormat(d3.format('d'));

	d3.select('#tracks-charts svg').datum(
		[{
			key: 'Tracks',
			values: formattedData
		}]
	).call(chart);

	nv.utils.windowResize(chart.update);

	return chart;
});
```
Do you think I am missing something here?

**Comment by timhoeppner** [comment]
Are you getting the nv is undefined error? Open up the console in Chrome and type "nv" and enter, is it defined? If it is then you have other issues.

**Comment by mayvn10** [comment]
Ok I'll log a different issue.

**Comment by mayvn10** [comment]
@timhoeppner reverting back fixed the nv issue

**Comment by cunnagin** [comment]
If you put the built source code files:

`nv.d3.js`
`nv.d3.css`

into your Meteor app directory (under a 'client' code folder) you can modify the 'nv.d3.js' file line:

`var nv = {};`

to

`var nv = window.nv || {};`

In order to get the global access. Has to be on client code access only or the window object will crash the server. User 'jlubrano' has submitted a pull request for fixing this using Meteor package/export.js changes I think. Would be much better for Meteor folks!

**Comment by liquidpele** [comment]
I merged in https://github.com/novus/nvd3/pull/1245

**Comment by liquidpele** [comment]
Will likely do a 1.8.2 release very soon.

**Comment by kublermdk** [comment]
I can confirm that following @cunnagin suggestion and instead of using the nvd3 Meteor package, putting the NVD3 files (CSS and JS) into a client folder managed to fix the issue. I didn't even need to adjust the var declaration.

**Comment by jstoup111** [comment]
Any status on when this will get released?

**Comment by liquidpele** [comment]
I need to do some manual testing first mostly...  Just been busy last couple weeks.

**Comment by jlukic** [comment]
Would love to see this fixed for meteor packages

**Comment by zocoi** [comment]
Currently I have to clone the repo in `packages` folder to make it work. Would love to release 1.8.2 soon.
Thanks

**Comment by Benco1** [comment]
Is this issue fixed? Looks like 1.8.1 is latest on atmosphere.
